nestkernel/recording_device.cpp

```diff
@@ -984,9 +989,17 @@ nest::RecordingDevice::build_filename_() const
     basename << node_.get_name();
   }
 
-  basename << "-" << std::setfill( '0' ) << std::setw( gidigits )
-           << node_.get_gid() << "-" << std::setfill( '0' )
-           << std::setw( vpdigits ) << node_.get_vp();
+  if ( P_.use_gid_ and not P_.label_.empty() )
+  {
+    basename << "-" << std::setfill( '0' ) << std::setw( gidigits )
+             << node_.get_gid() << "-" << std::setfill( '0' )
+             << std::setw( vpdigits ) << node_.get_vp();
+  }
+  else
```

Comment:
`not P_.label.empty()` needs to be part of the else branch. Perhaps reverse the order.